src/commands/prepare.ts

```diff
@@ -370,35 +376,81 @@ async function checkChangelog(
       `Changelog policy is set to "${changelogPolicy}", nothing to do.`
     );
     return;
-  } else if (changelogPolicy === ChangelogPolicy.Simple) {
-    logger.info('Checking the changelog...');
-    logger.debug(`Changelog policy: "${changelogPolicy}".`);
-    const relativePath = relative('', changelogPath);
-    if (relativePath.startsWith('.')) {
-      throw new ConfigurationError(
-        `Invalid changelog path: "${changelogPath}"`
-      );
-    }
-    if (!existsSync(relativePath)) {
-      throw new ConfigurationError(
-        `Changelog does not exist: "${changelogPath}"`
-      );
-    }
-    const changelogString = readFileSync(relativePath).toString();
-    logger.debug(`Changelog path: ${relativePath}`);
-    const changeset = findChangeset(changelogString, newVersion);
-    if (!changeset || !changeset.body) {
-      throw new ConfigurationError(
-        `No changelog entry found for version "${newVersion}"`
-      );
-    }
-    logger.debug(`Changelog entry found:\n"""\n${changeset.body}\n"""`);
-    return;
-  } else {
+  } else if (
+    changelogPolicy !== ChangelogPolicy.Auto &&
+    changelogPolicy !== ChangelogPolicy.Simple
+  ) {
     throw new ConfigurationError(
       `Invalid changelog policy: "${changelogPolicy}"`
     );
   }
+
+  logger.info('Checking the changelog...');
+  logger.debug(`Changelog policy: "${changelogPolicy}".`);
+
+  const relativePath = relative('', changelogPath);
+  if (relativePath.startsWith('.')) {
+    throw new ConfigurationError(`Invalid changelog path: "${changelogPath}"`);
+  }
+
+  if (!existsSync(relativePath)) {
+    throw new ConfigurationError(
+      `Changelog does not exist: "${changelogPath}"`
+    );
+  }
+
+  let changelogString = (await fsPromises.readFile(relativePath)).toString();
```

Comment:
Just out of curiosity - is there a reason to switch to the promise version over `readFileSync`?

Reply:
Just want to keep the good hygiene of doing all I/O async.